include/sparrow/array/array_data_factory.hpp

```diff
@@ -487,21 +489,45 @@ namespace sparrow
         }
     }
 
+
     /**
      * \brief Creates a default array_data object with the specified layout and values.
      *
      * @tparam Layout The layout of the array_data object.
      * @tparam ValueRange The type of the input range for the values.
      * @param values The input range of values for the array_data object.
-     * @param bitmap The bitmap type for the array_data object.
-     * @param offset The offset for the array_data object.
      * @return A new array_data object with the specified layout, values, bitmap, and offset.
      */
     template <arrow_layout Layout, std::ranges::input_range ValueRange>
-        requires(!mpl::constant_range<ValueRange>)
     array_data
-    make_default_array_data(ValueRange&& values, const array_data::bitmap_type& bitmap, std::int64_t offset)
+    make_default_array_data(ValueRange&& values)
+    {
+        const std::size_t size = std::ranges::size(values);
+        array_data::bitmap_type bitmap(size, true);
+        const std::int64_t offset = 0;
+        return make_default_array_data<Layout>(std::forward<ValueRange>(values), std::move(bitmap), offset);
```

Comment:
Could be all constexpr if we change dynamic_bitset to constexpr too. Maybe add a todo.

Reply:
are you suggesting to make the whole `make_default_array_data` function and everything this function calls constexpr?

Reply:
yes but it requires to change dynamic_bitset class which is not the point of this PR. So maybe just write a TODO comment and we will do that later.

Reply:
In that case I prefer to add an new issue instead of TODOs in the code

Reply:
https://github.com/man-group/sparrow/issues/166